### PR TITLE
aglae changes and footer w price

### PR DESCRIPTION
**Pull request by diarmuidmcg**

### CODE REVIEW

File: .idea/deploymentTargetDropDown.xml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="deploymentTargetDropDown">
```
Comment (ThibaultMiam) [inline .idea/deploymentTargetDropDown.xml]: to remove

File: .idea/gradle.xml [hunk 2]
```diff
@@ -5,16 +5,16 @@
     <option name="linkedExternalProjectsSettings">
       <GradleProjectSettings>
         <option name="testRunner" value="GRADLE" />
-        <option name="distributionType" value="DEFAULT_WRAPPED" />
         <option name="externalProjectPath" value="$PROJECT_DIR$" />
-        <option name="gradleJvm" value="11" />
+        <option name="gradleJvm" value="jbr-17" />
```
Comment (ThibaultMiam) [inline .idea/gradle.xml]: to remove and add to git ignore

File: .idea/migrations.xml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ProjectMigrations">
```
Comment (ThibaultMiam) [inline .idea/migrations.xml]: remove too

File: CoursesUUI/build.gradle.kts [hunk 4]
```diff
@@ -35,7 +35,7 @@ android {
 }
 
 dependencies {
-    api ("tech.miam.sdk:kmm-miam-sdk:4.0.0")
+    api ("tech.miam.sdk:kmm-miam-sdk:4.0.4")
```
Comment (ThibaultMiam) [inline CoursesUUI/build.gradle.kts]:
```suggestion
    api ("tech.miam.sdk:kmm-miam-sdk:4.0.7")
```

File: CoursesUUI/src/main/java/tech/miam/coursesuui/component/EmptyPage.kt [hunk 5]
```diff
@@ -2,6 +2,7 @@ package tech.miam.coursesuui.component
 
 import androidx.compose.foundation.Image
 import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
```
Comment (ThibaultMiam) [inline CoursesUUI/src/main/java/tech/miam/coursesuui/component/EmptyPage.kt]: rename the file too

File: CoursesUUI/src/main/java/tech/miam/coursesuui/config/MiamTemplateManager.kt [hunk 6]
```diff
@@ -221,6 +229,7 @@ class MiamTemplateManager {
                         }
                     }
                 }
+                empty { view = CoursesUEmptyPage() }
```
Comment (ThibaultMiam) [inline CoursesUUI/src/main/java/tech/miam/coursesuui/config/MiamTemplateManager.kt]: you can define CoursesUEmptyPage as generic if you put it in defautlt component config https://miamtech.github.io/mealz-documentation/docs/android/features/base-component/customize-views#empty

File: CoursesUUI/src/main/java/tech/miam/coursesuui/template/mealPlanner/meals/MealPlannerRecipeCardMealsListU.kt [hunk 7]
```diff
@@ -29,15 +27,16 @@ import androidx.compose.ui.unit.dp
 import coil.compose.AsyncImage
 import com.miam.core.localisation.Localisation
 import tech.miam.coursesuui.theme.Typography
```
Comment (ThibaultMiam) [inline CoursesUUI/src/main/java/tech/miam/coursesuui/template/mealPlanner/meals/MealPlannerRecipeCardMealsListU.kt:29]: rename file too

File: CoursesUUI/src/main/java/tech/miam/coursesuui/template/myMeal/CoursesUMyMealButton.kt [hunk 8]
```diff
@@ -42,12 +45,17 @@ class CoursesUMyMealButton: MyMealButtonSuccess {
...
+                contentColor = Colors.white,
             ) {
-
             Row(
                 horizontalArrangement = Arrangement.SpaceBetween,
-                verticalAlignment = Alignment.CenterVertically
+                verticalAlignment = Alignment.CenterVertically,
+                modifier = Modifier
+                    .clickable {
+                        println("MEALZDEBUG: click row")
```
Comment (ThibaultMiam) [inline CoursesUUI/src/main/java/tech/miam/coursesuui/template/myMeal/CoursesUMyMealButton.kt]: to remove

File: CoursesUUI/src/main/java/tech/miam/coursesuui/template/myMeal/CoursesUMyMealEmpty.kt [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+import com.miam.core.localisation.Localisation
+import com.miam.kmm_miam_sdk.android.theme.Colors
+import com.miam.kmm_miam_sdk.android.theme.Typography
+import com.miam.kmm_miam_sdk.android.ui.components.common.Clickable
+import com.miam.sdk.components.baseComponent.emptyPage.EmptyPage
+import com.miam.sdk.components.baseComponent.emptyPage.EmptyPageParameters
+import tech.miam.coursesuui.R
+
+class CoursesUMyMealAndFavoritesEmpty: EmptyPage {
+    @Composable
```
Comment (ThibaultMiam) [inline CoursesUUI/src/main/java/tech/miam/coursesuui/template/myMeal/CoursesUMyMealEmpty.kt]: do we need it if we have the default one ?

File: CoursesUUI/src/main/java/tech/miam/coursesuui/template/recipeCard/CoursesUCatalogCategoryRecipeCard.kt [hunk 10]
```diff
@@ -98,15 +98,15 @@ fun CoursesUCatalogCategoryRecipeCard(params: RecipeCardSuccessParams) {
                     horizontalArrangement = Arrangement.SpaceBetween,
                     verticalAlignment = Alignment.CenterVertically
                 ) {
-                    Box(modifier = Modifier.width(100.dp)) {
+                    Box() {
```
Comment (ThibaultMiam) [inline CoursesUUI/src/main/java/tech/miam/coursesuui/template/recipeCard/CoursesUCatalogCategoryRecipeCard.kt]:
```suggestion
                    Box {
```

File: CoursesUUI/src/main/res/values/plurals.xml [hunk 11]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+<!--    <plurals name="com_miam_my_meals_meal_in_basket">-->
```
Comment (ThibaultMiam) [inline CoursesUUI/src/main/res/values/plurals.xml]: it's used ?

File: app/google-services.json [hunk 12]
```diff
@@ -0,0 +1,190 @@
+{
```
Comment (ThibaultMiam) [inline app/google-services.json]: to remove add to git ignore

**Review by ThibaultMiam — CHANGES_REQUESTED** [review]

File: CoursesUUI/src/main/java/tech/miam/coursesuui/template/myMeal/CoursesUMyMealEmpty.kt (hunk 9 above)
Comment (diarmuidmcg) [reply CoursesUUI/src/main/java/tech/miam/coursesuui/template/myMeal/CoursesUMyMealEmpty.kt]: removed

File: .idea/render.experimental.xml
Comment (ThibaultMiam) [inline .idea/render.experimental.xml:1]: can you delete those files ?

**Review by ThibaultMiam — APPROVED** [review]

**Pull request merged**